engine/table/src/main/java/io/deephaven/engine/table/impl/sources/aggregate/SlicedCharAggregateColumnSource.java

```diff
@@ -0,0 +1,176 @@
+
+        if (startPos == NULL_LONG || endPos == NULL_LONG) {
+            return null;
+        }
+
+        final RowSet bucketRowSet = getPrevGroupRowSet(rowKey);
+        final long rowPos = bucketRowSet.find(rowKey);
+
+        final long size = bucketRowSet.size();
+        final long start = ClampUtil.clampLong(0, size, rowPos + startPos);
+        final long end = ClampUtil.clampLong(0, size , rowPos + endPos + 1);
+
+        // Determine the slice of the groupRowSetSource from start to end.
+        final RowSet rowSetSlice = bucketRowSet.subSetByPositionRange(start, end);
+        return rowSetSlice.isEmpty() ? null : new PrevCharVectorColumnWrapper(aggregatedSource, rowSetSlice);
+    }
+
+    @Override
+    public void fillChunk(@NotNull final FillContext context, @NotNull final WritableChunk<? super Values> destination,
+            @NotNull final RowSequence rowSequence) {
+        AggregateSlicedFillContext ctx = (AggregateSlicedFillContext) context;
+
+        final LongChunk<OrderedRowKeys> keyChunk = rowSequence.asRowKeyChunk();
+        final ObjectChunk<RowSet, ? extends Values> groupRowSetChunk = groupRowSetSource
+                .getChunk(ctx.groupRowSetGetContext, rowSequence).asObjectChunk();
+        final LongChunk<? extends Values> startChunk = startSource != null ?
+                startSource.getChunk(ctx.startGetContext, rowSequence).asLongChunk()
+                : null;
+        final LongChunk<? extends Values> endChunk = endSource != null ?
+                endSource.getChunk(ctx.endGetContext, rowSequence).asLongChunk()
+                : null;
+
+        final WritableObjectChunk<CharVector, ? super Values> typedDestination = destination.asWritableObjectChunk();
+        final int size = rowSequence.intSize();
+        for (int di = 0; di < size; ++di) {
+            // Transition from revTicks that include the current row to strict position offsets.
+            final long startPos = startChunk != null ? startChunk.get(di) : startOffset;
+            final long endPos = endChunk != null ? endChunk.get(di) : endOffset;
+
+            if (startPos == NULL_LONG || endPos == NULL_LONG) {
+                typedDestination.set(di, null);
+            } else {
+                final long rowKey = keyChunk.get(di);
+                final RowSet bucketRowSet = groupRowSetChunk.get(di);
+                final long rowPos = bucketRowSet.find(rowKey);
+
+                final long rowSetSize = bucketRowSet.size();
+                final long start = ClampUtil.clampLong(0, rowSetSize, rowPos + startPos);
+                final long end = ClampUtil.clampLong(0, rowSetSize , rowPos + endPos + 1);
+
+                // Determine the slice of the groupRowSetSource from start to end.
+                final RowSet rowSetSlice = bucketRowSet.subSetByPositionRange(start, end);
+                typedDestination.set(di, rowSetSlice.isEmpty() ? null : new CharVectorColumnWrapper(aggregatedSource, rowSetSlice));
```

Comment:
We need to think about empty or null here.

Reply:
Returning empty vectors now.